src/components/CommunityEvents.tsx

```diff
@@ -0,0 +1,256 @@
+//Libraries
+import React, { useEffect, useState } from "react"
+import { useI18next, useTranslation } from "gatsby-plugin-react-i18next"
+import {
+  Box,
+  Center,
+  Divider,
+  Flex,
+  Heading,
+  Icon,
+  Text,
+} from "@chakra-ui/react"
+import { FaDiscord } from "react-icons/fa"
+import { DateTime } from "luxon"
+
+// Components
+import ButtonLink from "./ButtonLink"
+import Link from "./Link"
+import Translation from "./Translation"
+
+// Constants
+import { GATSBY_FUNCTIONS_PATH } from "../constants"
+
+// Utils
+import { getData } from "../utils/cache"
+
+interface Event {
+  date: string
+  title: string
+  calendarLink: string
+  pastEventLink: string | undefined
+}
+
+interface State {
+  pastEventData: Array<Event>
+  upcomingEventData: Array<Event>
+  loading: boolean
+  hasError: boolean
+}
+
+const DiscordButton = () => {
+  return (
+    <ButtonLink to={"/discord/"} gap={2}>
+      <Icon as={FaDiscord} fontSize={25} />
+      Join Discord
+    </ButtonLink>
+  )
+}
+
+const CommunityEvents = () => {
+  const { language } = useI18next()
+  const { t } = useTranslation()
+  const [state, setState] = useState<State>({
+    pastEventData: [],
+    upcomingEventData: [],
+    loading: true,
+    hasError: false,
+  })
+
+  useEffect(() => {
+    try {
+      const fetchCalendarData = async () => {
+        const events = await getData(`${GATSBY_FUNCTIONS_PATH}/calendarEvents`)
+        const pastEventData = events.pastEvents.map((event) => {
+          return {
+            date: event.start.dateTime,
+            title: event.summary,
+            calendarLink: event.htmlLink,
+            pastEventLink: event.location,
+          }
+        })
+        const upcomingEventData = events.futureEvents.map((event) => {
+          return {
+            date: event.start.dateTime,
+            title: event.summary,
+            calendarLink: event.htmlLink,
+            pastEventLink: event.location,
+          }
+        })
+        setState({
+          ...state,
+          pastEventData,
+          upcomingEventData,
+          loading: false,
+          hasError: false,
+        })
+      }
+      fetchCalendarData()
+    } catch {
+      setState({ ...state, loading: false, hasError: true })
+    }
+  }, [])
+
+  const renderEventDateTime = (
```

Comment:
Define this function outside the component definition. It is a pure function and there is no reason to have it inside as it is going to be created every time the component renders.